eng/targets/Helix.targets

```diff
@@ -16,6 +16,16 @@
     <HelixPreCommand Include="call RunPowershell.cmd InstallNode.ps1 $(NodeVersion) %25HELIX_CORRELATION_PAYLOAD%25\node\bin || exit /b 1" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(IsHelixJob)' == 'true' AND '$(IsWindowsHelixQueue)' == 'true' AND '$(TestDependsOnAspNetRef)' == 'true'">
+    <HelixContent Include="$(RepoRoot)artifacts\packages\Release\Shipping\Microsoft.AspNetCore.App.Ref.$(AppRuntimeVersion).nupkg" />
+    <HelixPreCommand Include="call RunPowershell.cmd InstallAspNetRef.ps1 Microsoft.AspNetCore.App.Ref.$(AppRuntimeVersion).nupkg Microsoft.AspNetCore.App.Ref || exit /b 1" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(IsHelixJob)' == 'true' AND '$(IsWindowsHelixQueue)' == 'false' AND '$(TestDependsOnAspNetRef)' == 'true'">
+    <HelixContent Include="$(RepoRoot)artifacts\packages\Release\Shipping\Microsoft.AspNetCore.App.Ref.$(AppRuntimeVersion).nupkg" />
+    <HelixPreCommand Include="./installaspnetref.sh Microsoft.AspNetCore.App.Ref.$(AppRuntimeVersion).nupkg Microsoft.AspNetCore.App.Ref || exit /b 1" />
+  </ItemGroup>
```

Comment:
Could just skip these tests in the servicing branches on helix at that point too?

Reply:
I haven't worked on #14832 so I haven't made the call yet. However, I think it'd make sense to unconditionally enable it in master. Though I'm not sure why we would disable them unconditionally in 3.1; why not make it conditional on whether the targeting pack is building?

More specific to the changes in Helix.targets, we should make it dependent on `IsTargetingPackBuilding` if possible.

Reply:
Not sure exactly what changes you suggesting, the only test that uses this flag for now (and most likely ever) will be the framework unit tests, does that somehow trigger IsTargetingPackBuilding being true?   Maybe that's why the version files missing since we are building it in an unexpected way?

Reply:
```
      - script: ./build.cmd -ci -all -pack -arch x64 -buildNative /p:ASPNETCORE_TEST_LOG_DIR=artifacts/log /bl:artifacts/log/helix.daily.build.x64.binlog
        displayName: Build shared fx
      # Build the x86 shared framework
      - script: .\restore.cmd -ci /p:BuildInteropProjects=true
        displayName: Restore
      - script: .\build.cmd -ci -NoRestore -test -projects eng\helix\helix.proj /p:IsHelixJob=true /p:IsHelixDaily=true /p:BuildAllProjects=true /p:BuildInteropProjects=true /p:BuildNative=true /p:RunTemplateTests=true /p:ASPNETCORE_TEST_LOG_DIR=artifacts/log -bl
        displayName: Run build.cmd helix target
```

Taken from the CI.yml, we just do these steps, and so the would IsTargetingPackBuilding ever be true out of these?

Reply:
The idea right now is that we'll always build ref packs on master so it's not a problem there. We will run into issues when 5.0 goes into servicing since IsTargetingPackBuilding will be set to false (unless we mess up). In those cases, we should disable the targeting pack tests. As such, we should add that condition to these itemgroups as a form of future proofing against servicing.

Reply:
Okay so just add IsTargetingPackBuilding  to these targest, I can do that

Reply:
Well I guess, i'm not sure what the purpose of adding this check to the target is, since the tests will still run on helix with that check, they just will 100% fail since the ref pack won't get installed.  While if the packs don't build, we would get a much easier to diagnose build failure rather than a runtime helix test failure

Reply:
Is there an easy way to have this target error/stop if true when IsTargetingPackBuilding is false?

Reply:
The tests should see the `$(IsTargetingPackBuilding)` value because @JunTaoLuo hooked that up in the test project. The test simply `return`s when the targeting pack is _not_ building.

But, we may have a bad merge in there somewhere. Double-check the tests.

Reply:
Regardless, the conditions here should check `$(IsTargetingPackBuilding)`

Reply:
I double-checked the tests in 'master'. They should automatically no-op (unconditionally succeed) once we go into servicing

Reply:
Yes the tests already have logic for handling whether the targeting pack is building. The scripts here need to handle it as well. We just need to ensure in the scenario where the targeting pack is not built, these scripts won't fail.

Reply:
Cool, that should do it then